src/components/Nav/NavBar/NavBar.js

```diff
@@ -0,0 +1,59 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import styled from 'styled-components';
+import DsnbLogo from './DsnbLogo/DsnbLogo';
+import SearchBox from './SearchBox/SearchBox';
+import KakaoLogin from './KakaoLogin/KakaoLogin';
+
+const NavBar = ({ setIsListModalVisible }) => {
+  const [isLogin, setIsLogin] = useState(false);
+  const [searchInputValue, setSearchInputValue] = useState('');
+  const [authorData, setAuthorData] = useState([]);
+  const [titleData, setTitleData] = useState([]);
+
+  const filteredAuthorData = authorData?.filter(data => {
+    return data.author.includes(searchInputValue);
+  });
+
+  const filteredTitleData = titleData?.filter(data => {
+    return data.title.includes(searchInputValue);
+  });
+
+  useEffect(() => {
+    const fetchTitleData = async () => {
+      const res = await axios.get('data/titleData.json');
+      setTitleData(res.data);
+    };
+    fetchTitleData();
+
+    const fetchAuthorData = async () => {
+      const res = await axios.get('data/authorData.json');
+      setAuthorData(res.data);
+    };
+    fetchAuthorData();
+  }, []);
```

Comment:
```suggestion
const fetchTitleData = async () => {
      const res = await axios.get('data/titleData.json');
      setTitleData(res.data);
};

const fetchAuthorData = async () => {
      const res = await axios.get('data/authorData.json');
      setAuthorData(res.data);
 };
    
  useEffect(() => {
    fetchTitleData();
    fetchAuthorData();
  }, []);
```
리팩터링 하시면서 이런식으로 함수 추출을 해보세요~!

Reply:
감동입니다..

Reply:
감사합니다 피드백 반영했습니다 🙆🏽‍♂️